internal/idle/idle_test.go

```diff
@@ -306,51 +309,59 @@ const (
 // racyIdlnessEnforcer is a test idleness enforcer used specifically to test the
 // race between idle timeout and incoming RPCs.
 type racyEnforcer struct {
-	state *racyState // Accessed atomically.
+	t       *testing.T
+	state   *racyState // Accessed atomically.
+	started bool
 }
 
 // ExitIdleMode sets the internal state to stateExitedIdle. We should only ever
 // exit idle when we are currently in idle.
 func (ri *racyEnforcer) ExitIdleMode() error {
+	// Set only on the initial ExitIdleMode
+	if ri.started == false {
+		if *ri.state != stateInitial {
```

Comment:
Don't we have to do a `atomic.Load(ri.State)` here?

Reply:
I believe if there _is_ a bug, then this would show up as a data race.  But yes, it'd be better to do it atomically, and best to do as a cmpswp, but expecting the previous state to be `Initial`.  Done.